src/main/scala/firrtl/graph/RenderDiGraph.scala

```diff
@@ -0,0 +1,233 @@
+  * @param graphName Name of the graph, when shown in viewer
+  * @param rankDir   Graph orientation, default is LR (left to right), most common alternative is TB (top to bottom)
+  * @tparam T        The type of the Node.
+  */
+class RenderDiGraph[T <: Any](diGraph: DiGraph[T], graphName: String = "", rankDir: String = "LR") {
+
+
+  /**
+    * override this to change the default way a node is displayed. Default is toString surrounded by double quotes
+    * {{{
+    * val rend = new RenderDiGraph(graph, "alice") {
+    *   override def renderNode(node: Symbol): String = s"\"${symbol.name}\""
+    *   }
+    * }}}
+    */
+  def renderNode(node: T): String = {
+    s""""${node.toString}""""
+  }
+
+  def findOneLoop: Set[T] = {
+    var path = Seq.empty[T]
+
+    try {
+      diGraph.linearize
+    }
+    catch {
+      case cyclicException: CyclicException =>
+        val node = cyclicException.node.asInstanceOf[T]
+        val pathFound = diGraph.getEdges(node).exists { vertex =>
+          try {
+            path = diGraph.path(vertex, node, blacklist = Set.empty)
+            true
+          }
+          catch {
+            case _: PathNotFoundException =>
+            case t: Throwable =>
+              throw t
+
+          }
+          false
+        }
+
+    }
+    path.toSet
+  }
+
+  /**
+    * Searches a DiGraph for a cycle. The first one found will be rendered as a graph that contains
+    * only the nodes in the cycle plus the neighbors of those nodes.
+    * @return a string that can be used as input to the dot command, string is empty if no loop
+    */
+  def showOnlyTheLoopAsDot: String = {
+    // finds the loop
+
+    val loop = findOneLoop
+
+    if(loop.nonEmpty) {
+
+      // Find all the children of the nodes in the loop
+      val childrenFound = diGraph.getEdgeMap.flatMap {
+        case (node, children) if loop.contains(node) => children
+        case _ => Seq.empty
+      }.toSet
+
+      // Create a new DiGraph containing only loop and direct children or parents
+      val edgeData = diGraph.getEdgeMap
+      val newEdgeData = edgeData.flatMap { case (node, children) =>
+        if(loop.contains(node)) {
+          Some(node -> children)
+        }
+        else if(childrenFound.contains(node)) {
+          Some(node -> children.intersect(loop))
+        }
+        else {
+          val newChildren = children.intersect(loop)
+          if(newChildren.nonEmpty) {
+            Some(node -> newChildren)
+          }
+          else {
+            None
+          }
+          }
+      }
+
+      val justLoop = DiGraph(newEdgeData)
+      val newRenderer = new RenderDiGraph(justLoop, graphName, rankDir) {
+        override def renderNode(node: T): String = {
+          super.renderNode(node)
+        }
+      }
+      newRenderer.toDotWithLoops(loop, getRankedNodes)
+    }
+    else {
+      ""
+    }
+  }
+
+  /**
+    * returns a string that is a graphviz digraph
+    * @return
```

Comment:
Maybe put the return in the return field?

Reply:
Done